### PR TITLE
(GH-50) change svn info settings default value of revision and depth

### LINKED ISSUE #50: Change SvnInfoSettings default value of revision and depth
SvnInfoSettings.Depth should be empty by default.
SvnInfoSettings.Revision should be nullable to allow default setting.

### PR DESCRIPTION
**Pull request by christianbumann**
Fixes #50

### CODE REVIEW

File: src/Cake.Svn.Tests/Unit/Info/SvnInfoResultTests.cs [hunk 1]
```diff
@@ -24,21 +24,6 @@ public void Should_Throw_If_RepositoryRoot_Is_Null()
...
-                    LastChangedAuthor = null
-                };
-
-                // When
-                // Then
-                Assert.Throws<ArgumentNullException>("lastChangedAuthor", () =>
-                    GetSvnInfoResultFromParameterList(testSvnInfoResult));
-            }
-
-            [Fact]
```
Comment (pascalberger) [inline src/Cake.Svn.Tests/Unit/Info/SvnInfoResultTests.cs]: Why is this removed?

File: src/Cake.Svn/Info/SvnInfoResult.cs [hunk 2]
```diff
@@ -31,7 +31,6 @@ public sealed class SvnInfoResult
             SvnKind nodeKind)
         {
             repositoryRoot.NotNull(nameof(repositoryRoot));
-            lastChangedAuthor.NotNull(nameof(lastChangedAuthor));
```
Comment (pascalberger) [inline src/Cake.Svn/Info/SvnInfoResult.cs]: Why?

File: src/Cake.Svn/Info/SvnInfoSettings.cs [hunk 3]
```diff
@@ -10,16 +10,16 @@ public sealed class SvnInfoSettings : SvnRemoteSettings
...
-            Depth = SvnDepth.Infinity;
             RetrieveActualOnly = true;
             RetrieveExcluded = true;
-            Revision = -1;
         }
 
         /// <summary>
         /// Gets or sets the value of the revision on which Subversion should operate.
+        /// Use <c>null</c> for the default revision, <c>-1</c> for the head revision and numbers 
+        /// bigger than 0 for a cartain revision.
```
Comment (pascalberger) [inline src/Cake.Svn/Info/SvnInfoSettings.cs]: What about 0?

File: src/Cake.Svn/Info/SvnInfoSettings.cs [hunk 4]
```diff
@@ -10,16 +10,16 @@ public sealed class SvnInfoSettings : SvnRemoteSettings
         /// </summary>
         public SvnInfoSettings()
         {
-            Depth = SvnDepth.Infinity;
```
Comment (pascalberger) [inline src/Cake.Svn/Info/SvnInfoSettings.cs]: What's the reason for this change?

File: src/Cake.Svn/Info/SvnInfoResult.cs (hunk 2 above)
Comment (christianbumann) [reply src/Cake.Svn/Info/SvnInfoResult.cs]: @pascalberger because the returned result for lastChangedAuthor can be null. Would it better pass string.Empty if null?

File: src/Cake.Svn/Info/SvnInfoSettings.cs (hunk 4 above)
Comment (christianbumann) [reply src/Cake.Svn/Info/SvnInfoSettings.cs]: @pascalberger Infinity is not the default on SvnInfoArgs. And if we get information from a file we only get one result for this file. Same should be for a directory. If we want more information about directory and the subdirectories/files inside we should explicit demand this with changing the new default to e.g. Infinity.

File: src/Cake.Svn/Info/SvnInfoSettings.cs (hunk 3 above)
Comment (christianbumann) [reply src/Cake.Svn/Info/SvnInfoSettings.cs]: Should bigger than -1 I will correct this

File: src/Cake.Svn.Tests/Unit/Info/SvnInfoSettingsTests.cs [hunk 5]
```diff
@@ -48,13 +48,13 @@ public void Should_Not_IncludeExternals_By_Default()
             }
 
             [Fact]
-            public void Should_Set_Revision_Minus_One_By_Default()
+            public void Should_Set_Null_By_Default()
```
Comment (pascalberger) [inline src/Cake.Svn.Tests/Unit/Info/SvnInfoSettingsTests.cs]: Should be called `Should_Set_Revision_To_Null_By_Default`

File: src/Cake.Svn/Internal/SharpSvnClient.cs [hunk 6]
```diff
@@ -113,7 +113,7 @@ public IEnumerable<SvnInfoResult> GetInfo(string repositoryUrl, SvnInfoSettings
                     select new SvnInfoResult(
                         eventArg.RepositoryId,
                         eventArg.RepositoryRoot,
-                        eventArg.LastChangeAuthor,
+                        eventArg.LastChangeAuthor ?? string.Empty,
```
Comment (pascalberger) [inline src/Cake.Svn/Internal/SharpSvnClient.cs]: Why? What's the problem with returning `null`? Beside this change has nothing to do this the linked issue, so I'm confused why this should be changed.

Comment (christianbumann) [reply src/Cake.Svn/Internal/SharpSvnClient.cs]: @pascalberger if we pass null to the ctor an Exception will be thrown as we check this parameter to null. LastChangeAuthor can be null if the file was added to the working copy but not yet committed. This case happens after change I did to nullable revision.

Comment (pascalberger) [reply src/Cake.Svn/Internal/SharpSvnClient.cs]: @christianbumann Wouldn't it be better in this case to allow LastChangeAuthor to be null in `SvnInfoResult` ctor?

Comment (christianbumann) [reply src/Cake.Svn/Internal/SharpSvnClient.cs]: @pascalberger yes this would be an option (I had this in the first commit of this PR) but changed to the string.Empty behaviour. If we change this to allow null, we should also allow null on FullPath...

**Comment by christianbumann** [comment]
@pascalberger I did the corrections on the second commit.
